src/panels/CreateFleetPanel.ts

```diff
@@ -0,0 +1,136 @@
+
+    constructor(
+        private readonly sessionProvider: ReadyAzureSessionProvider,
+        private readonly subscriptionId: string,
+        private readonly subscriptionName: string,
+        // commandId: string,
+    ) {
+        this.resourceManagementClient = getResourceManagementClient(sessionProvider, this.subscriptionId);
+        this.fleetClient = getAksFleetClient(sessionProvider, this.subscriptionId);
+        this.featureClient = getFeatureClient(sessionProvider, this.subscriptionId);
+        // this.commandId = commandId;
+        console.log(this.resourceManagementClient, this.featureClient, this.sessionProvider);
+    }
+
+    getTitle(): string {
+        return "Create Fleet";
+    }
+
+    getInitialState(): InitialState {
+        return {
+            subscriptionId: this.subscriptionId,
+            subscriptionName: this.subscriptionName,
+        };
+    }
+
+    getTelemetryDefinition(): TelemetryDefinition<"createFleet"> {
+        return {
+            getResourceGroupsRequest: false,
+            getLocationsRequest: false,
+            createFleetRequest: true,
+        };
+    }
+
+    getMessageHandler(webview: MessageSink<ToWebViewMsgDef>): MessageHandler<ToVsCodeMsgDef> {
+        return {
+            getLocationsRequest: () => this.handleGetLocationsRequest(webview),
+            getResourceGroupsRequest: () => this.handleGetResourceGroupsRequest(webview),
+            createFleetRequest: (args) =>
+                this.handleCreateFleetRequest(
+                    args.isNewResourceGroup,
+                    args.resourceGroupName,
+                    args.location,
+                    args.name,
+                ),
+            // args.isNewResourceGroup,
+            // args.resourceGroupName,
+            // args.location,
+            // args.name,
+            // args.hubClusterMode,
+            // webview,
+        };
+    }
+
+    private handleGetLocationsRequest(webview: MessageSink<ToWebViewMsgDef>) {
+        console.log(webview);
+    }
+
+    private handleGetResourceGroupsRequest(webview: MessageSink<ToWebViewMsgDef>) {
+        console.log(webview);
+    }
+
+    private async handleCreateFleetRequest(
+        isNewResourceGroup: boolean,
+        resourceGroupName: string,
+        location: string,
+        name: string,
+        // hubClusterMode: HubClusterMode,
+        // webview: MessageSink<ToWebViewMsgDef>,
+    ) {
+        if (isNewResourceGroup) {
+            // create new resource group
+        }
+
+        const resource = {
+            location: location,
+        };
+
+        try {
+            await createFleet(this.fleetClient, resourceGroupName, name, resource);
+        } catch (error) {
+            console.log(error);
+        }
+    }
+}
+
+async function createFleet(
+    client: ContainerServiceFleetClient,
+    groupName: string,
+    name: string,
+    resource: Fleet,
+    // sessionProvider: ReadyAzureSessionProvider,
+    // subscriptionId: string,
```

Comment:
the client is created using the session and the subscriptionId, so I don't think that you would ever need these parameters here.
if the list of argument increases (probably, to support all the fleet options) then you should create a type to encapsulate them. That way, the function signature won't change when you add more options

Reply:
I feel this PR is still undergoing changes, testing etc.. it could be still work in progress. Please reach out if you have any questions, Thank you.

Reply:
It is absolutely work in progress, I'm commenting to help get to a mergeable state in a reasonable time :)
Thanks for reviewing as well @hsubramanianaks!

Reply:
@serbrech agreed to your point above where it has to be a type for the function parameter.